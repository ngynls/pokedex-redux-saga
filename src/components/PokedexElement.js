import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import '../components/PokedexElement.scss';
import PokedexUtils from '../components/shared/PokedexUtils';

class PokedexElement extends Component {

    render(){
        const id=this.props.id;
        const name= PokedexUtils.capitalizeName(this.props.pokemon.name);
        const spriteUrl=`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png`;

        return (
            <Link className="card" to={`/pokemon/${id}`}>
                <img src={spriteUrl} className="card-img-top sprite-icon" alt="sprite" />
                <div className="card-body">
                    <p className="card-text">{name}</p>
                </div>
            </Link>
        );
    }
}

export default PokedexElement;