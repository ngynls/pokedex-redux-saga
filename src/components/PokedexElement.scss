.card{
    margin-top:1rem;
    margin-bottom:1rem;
}

.card-text{
    text-align:center;
}