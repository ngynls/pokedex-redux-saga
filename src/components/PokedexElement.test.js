import React from "react";
import renderer from "react-test-renderer";
import PokedexElement from "./PokedexElement";
import { BrowserRouter as Router } from "react-router-dom";

test("matches snapshot of PokedexElement", () => {
    const id = 0;
    const pokemon = {
        name: "bulbasaur"
    };
    const tree = renderer.create(<Router><PokedexElement id={ id + 1 } pokemon={ pokemon }/></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});