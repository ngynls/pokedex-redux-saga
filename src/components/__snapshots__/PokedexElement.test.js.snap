// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of PokedexElement 1`] = `
<a
  className="card"
  href="/pokemon/1"
  onClick={[Function]}
>
  <img
    alt="sprite"
    className="card-img-top sprite-icon"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
  />
  <div
    className="card-body"
  >
    <p
      className="card-text"
    >
      Bulbasaur
    </p>
  </div>
</a>
`;
